### DISCUSSION TITLE
Can developing devices cause a home to become unreliable or even corrupt?

### DISCUSSION THREAD

**Discussion by jankornblum** [HomeSpan/HomeSpan#484] [Q&A]
When developing custom devices, might this cause a home to get slow or unreliable or even corrupt? Especially when unpairing a device from console, changing a devices id or accessory id or similar things... Can this lead to something like a corrupted state, and homekit (hubs) thinks there are still any devices which don't exist any longer? Or will homekit "repair" itself and remove outdated or non longer existing devices?

What do i have to take care about in detail? Should i better use a dedicated "development" home at all?

**Comment by HomeSpan** [comment]
I have not encountered or heard of any issues where HomeSpan can lead to instabilities in your overall HomeKit network, though it certainly is the case that as you develop and test a HomeSpan sketch it's fairly easy to confuse HomeKit about your specific Accessory.  For example, if you add a Service or Characteristic, the Home App may not render it correctly, or it may give the dreaded "No Response" message.  Sometimes after a few minutes the Home App sorts itself out and everything works as expected.  But often the only way to fix the issue is to:

* Unpair the Accessory from the Home App
* Type 'H' into the Serial Monitor to clear all pairing information and the Device ID from your Accessory
* Re-Pair the Accessory with the Home App

HomeKit identifies Accessories according to their Device ID, which is supposed to be fixed in every commercial Accessory.  HomeSpan randomly generates a HomeKit Device ID when it is first run on a new ESP32 and saves the value in non-volatile storage.  Typing 'H' deletes the saved entry so that the next time HomeSpan runs it generates a new Device ID that causes HomeKit to think it is a completely new device.

**Reply by jankornblum** [reply]
@HomeSpan Thx for clarifying in detail! And apologize, my question was rather meant like „can wrong usage of HomeSpan cause any problems“, hope it didn’t read as „can HomeSpan cause“ ;) 

So independent from HomeSpan (but good to know for further developing using it): There cannot exist any abandoned devices (caused by wrong or incomplete unpairing or similar) which are not visible in home app, but despite still exist invisible on the hubs, correct?
